### PR TITLE
Merging of Dev (Assignment 3) into Master

### PR DESCRIPTION
**Pull request by anthonynmh**
This PR consists of:

Using OAuth and JWT to implement authentication and authorization functionality
The Auth functionality allows for:
a. A designated maintainer role whose privileges allow users to update and delete the questions
in the question repository.
b. Registered users should be allowed to read the questions from the question repository.
c. Unauthenticated and unauthorized users should not be allowed to access questions (either
for reading or updating).

### CODE REVIEW

**Review by cadencjk — CHANGES_REQUESTED** [review]
Need to remove line 23 from README

File: backend/server/repositories/userRepoMySql.js [hunk 1]
```diff
@@ -82,8 +83,10 @@ const getAllUserInfo = async () => {
 };
 
 const getUserInfoById = async (userId) => {
-  var _userInfo = {};
+  // idk why this is needed but there was error in the SQL syntax here if i dont do this
+  // userId = userId['_userId'];
```
Comment (venuslimm) [inline backend/server/repositories/userRepoMySql.js:87]: I think these should be removed

File: package.json
[the remark this answers is not in the document]
Comment (venuslimm) [reply package.json:1]: Yes, it needs to be. README.md needs to be updated if this modification is made to package.json

**Review by venuslimm — CHANGES_REQUESTED** [review]

**Review by JrmCkh — APPROVED** [review]
LGTM

File: package.json
[the remark this answers is not in the document]
Comment (anthonynmh) [reply package.json:1]: Updated readme and package.json

File: README.md [hunk 2]
```diff
@@ -20,67 +20,67 @@ For development, you may also want to install:
 1. Search for **MySQL 8.1 Command Line Client** on your computer. Execute it to open up the terminal.
 2. Enter your root password.
 3. Run the SQL statements in `./backend/schema.sql` on the terminal.
+4. Run the SQL statements in `./Auth/schema.sql` on the terminal.
```
[the remark this answers is not in the document]
Comment (anthonynmh) [reply README.md]: removed

File: backend/server/repositories/userRepoMySql.js (hunk 1 above)
Comment (anthonynmh) [reply backend/server/repositories/userRepoMySql.js]: removed

**Review by cadencjk — APPROVED** [review]
LGTM

**Review by venuslimm — APPROVED** [review]
LGTM

**Pull request merged**